tracer/time_windows_test.go

```diff
@@ -0,0 +1,30 @@
+package tracer
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func BenchmarkNormalTimeNow(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		lowPrecisionNow()
+	}
+}
+
+func BenchmarkHighPrecisionTime(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		highPrecisionNow()
+	}
+}
+
+func TestHighPrecisionTimerIsMoreAccurate(t *testing.T) {
+	startLow := lowPrecisionNow()
```

Comment:
This is flaky, but as you say above, it's hard to avoid. I think you could reduce the flakiness by  something like : 

```
for startLow == lowPrecisionNow() {} // make sure we're just after a "low precision boundary"
startLow = lowPrecisionNow()
```